### PR TITLE
feat(gateway): deprecate the contactPoint property and introduce the initialContactProperties

### LINKED ISSUE #4856: Support mutliple cluster contact points on the standalone Gateway
**Is your feature request related to a problem? Please describe.**

Currently we support only one cluster contact point, which is problematic when this contact point is currently not available. 
This can happen in multiple situations, directly on start but also on a network partition. Imagine the gateway was partitioned it removes all there nodes, because it can't connect to them. When the network partition is over but the contact point is not available it will not be able to connect to the rest of the cluster.

We handle this in k8s setup via dns, since we use as contact point a domain which can be resolved to multiple nodes. This is not easy possible on bare metal, docker setups or integration tests.

Especially on the last one I would like to have these kind of feature, because it makes it possible to write more different integration tests and make them also less flaky.

It is also a bit confusing why we support only one in the gateway and multiple at the broker side.

_Note: The contact point is directly used as initial contact point in atomix._

**Describe the solution you'd like**

Similar to the initial contact points in the broker I would expect a list of contact points.

**Describe alternatives you've considered**
Idk how to fix that in it tests or docker compose setup?

### PR DESCRIPTION
**Pull request by aivinog1**
## Description

Replace the `contactPoint` property with the `initialContactPoints` property. So, there now could be an array of broker connections on the start.

## Related issues

closes #4856

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [X] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [X] There are unit/integration tests that verify all acceptance criterias of the issue
* [X] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by saig0** [comment]
@Zelldon can you have a look at the PR?

File: dist/src/main/config/gateway.yaml.template [hunk 1]
```diff
@@ -48,6 +48,17 @@
       # minKeepAliveInterval: 30s
 
     # cluster:
+      # Sets initial contact points of brokers
```
Comment (Zelldon) [inline dist/src/main/config/gateway.yaml.template:51]:
```suggestion
      # Sets initial contact points (brokers), which the gateway should contact to
```

File: dist/src/main/config/gateway.yaml.template [hunk 2]
```diff
@@ -48,6 +48,17 @@
...
+      # The contact points of the internal network configuration must be specified.
+      # The format is [HOST:PORT]
+      # Example:
+      # initialContactPoints : [ 192.168.1.22:26502, 192.168.1.32:26502 ]
+      # This setting can also be overridden using the environment variable ZEEBE_GATEWAY_CLUSTER_INITIALCONTACTPOINTS
+      # specifying a comma-separated list of contact points.
+      # Default is 127.0.0.1:26502
```
Comment (Zelldon) [inline dist/src/main/config/gateway.yaml.template:58]: This is duplicated with the next line
```suggestion
```

**Review by ChrisKujawa — COMMENTED** [review]
Awesome @aivinog1 thanks for your contribution!

I have just some smaller comments, nothing major! After fixing these we can merge it. :rocket:

**Comment by aivinog1** [comment]
Hey @Zelldon! It's done :)

**Review by ChrisKujawa — APPROVED** [review]
Thanks @aivinog1 :rocket:

bors r+

**Pull request merged**